roles/core/nic/templates/ifcfg.j2

```diff
@@ -1,53 +1,59 @@
+#jinja2: lstrip_blocks: "True"
 #### Blue Banquise file ####
 ## {{ansible_managed}}
 
 {# 1. First the name and default parameters. #}
 NAME={{item}}
 DEVICE={{item}}
-{% if (network_interfaces[item]['autoconnect'] is defined and not none) and (network_interfaces[item]['autoconnect'] == false) %}
+{% if (network_interfaces[item]['autoconnect'] is defined and network_interfaces[item]['autoconnect'] is not none) and (network_interfaces[item]['autoconnect'] == false) %}
 ONBOOT=no
 {% else %}
 ONBOOT=yes
 {% endif %}
+{% if ansible_facts.distribution_major_version < 8 %}
 NM_CONTROLLED={{nic_nm_controlled | ternary('yes','no')}}
+{% endif %}
 BOOTPROTO=none
 
 {# 2. Let's get the type. Default is Ethernet. #}
 {% set nic_type = 'Ethernet' %}
 {% if network_interfaces[item]['type'] is defined and not none %}
-{% if network_interfaces[item]['type'] == 'bond' %}
-{% set nic_type = 'Bond' %}
-{% elif network_interfaces[item]['type'] == 'infiniband' %}
-{% set nic_type = 'Infiniband' %}
+  {% if network_interfaces[item]['type'] == 'bond' %}
+    {% set nic_type = 'Bond' %}
+  {% elif network_interfaces[item]['type'] == 'infiniband' %}
+    {% set nic_type = 'Infiniband' %}
+  {% endif %}
 {% endif %}
+{% if (network_interfaces[item]['vlan'] is defined) and (network_interfaces[item]['vlan'] == true) %}
+  {% set nic_type = 'Vlan' %}
 {% endif %}
 TYPE={{nic_type}}
```

Comment:
Hi @oxedions,

Proposal :

```
{% if (network_interfaces[item]['type'] is defined) %}
  {% if (network_interfaces[item]['type'] | lower) in ['infiniband', 'bond', 'vlan'] %}
TYPE={{ network_interfaces[item]['type'] | lower }}
  {% endif %}
{% else %}
TYPE=ethernet
{% endif %}
```

"and not none" instruction on this line isn't taken into account (must be "and var is not none") and not necessary, you fill TYPE with ethernet even if this variable is "none"

```
{% if network_interfaces[item]['type'] is defined and not none %}
```

You also make a difference between 'vlan' type and the others types of nic. Having a 'vlan' var in your network_interfaces dict is a little disturbing (sounds like we must fill this var with a VLAN number)

Be careful, I'm not sure type is case insensitive, only tested on RH8 and "vlan" type.

Reply:
The `vlan` var in the network_interfaces matches the `VLAN=yes|no` var from sysconfig.

Reply:
I tried to match sysconfig strictly. But I also understand your concern.

Question is: can it happen (not sure) that a configuration is using type != vlan but VLAN=yes ?

Reply:
It seems not. I only find [one place in the code](https://github.com/NetworkManager/NetworkManager/blob/master/src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-writer.c#L1291) where NetworkManager writes VLAN=yes in the ifcfg file. This is the master branch, I did not check the sources of EL7/8.

Reply:
https://github.com/NetworkManager/NetworkManager/commit/e8ffca398f429e2d8ac4e2131c9aa54d13fb3fbf

Reply:
The NM ifcfg reader condition is not case sensitive: `(!strcasecmp (type, TYPE_VLAN))`